src/app/shell/DefaultAccount.tsx

```diff
@@ -1,48 +1,79 @@
 import { getPlatforms } from 'app/accounts/platforms';
-import { accountsLoadedSelector, currentAccountSelector } from 'app/accounts/selectors';
+import SelectAccount from 'app/accounts/SelectAccount';
+import {
+  accountsLoadedSelector,
+  accountsSelector,
+  currentAccountMembershipIdSelector,
+  destinyVersionSelector,
+} from 'app/accounts/selectors';
 import ShowPageLoading from 'app/dim-ui/ShowPageLoading';
 import { t } from 'app/i18next-t';
 import { accountRoute } from 'app/routes';
 import { useThunkDispatch } from 'app/store/thunk-dispatch';
 import { RootState } from 'app/store/types';
-import React, { useEffect } from 'react';
+import { useEffect } from 'react';
 import { useSelector } from 'react-redux';
 import { Navigate } from 'react-router';
 import ErrorPanel from './ErrorPanel';
 
 /**
- * A view for when there's been an error loading accounts or there are no accounts.
+ * DefaultAccount handles when there is no URL path selecting a specific
+ * account. It attempts to redirect to the last used account, or otherwise shows
+ * either a menu of accounts or an error.
  */
 export default function DefaultAccount() {
   const dispatch = useThunkDispatch();
-  const activeAccount = useSelector(currentAccountSelector);
+  const accounts = useSelector(accountsSelector);
   const accountsLoaded = useSelector(accountsLoadedSelector);
   const accountsError = useSelector((state: RootState) => state.accounts.accountsError);
 
+  const currentAccountMembershipId = useSelector(currentAccountMembershipIdSelector);
+  const destinyVersion = useSelector(destinyVersionSelector);
+
   useEffect(() => {
-    if (!accountsLoaded) {
+    // If currentAccountMembershipId is set we'll redirect immediately, we don't need to load accounts
+    if (!accountsLoaded && !currentAccountMembershipId) {
       dispatch(getPlatforms());
     }
-  }, [dispatch, accountsLoaded]);
+  }, [dispatch, accountsLoaded, currentAccountMembershipId]);
+
+  // Show a loading screen while we're still loading accounts.
+  // If currentAccountMembershipId is set we'll redirect immediately, we don't need to load accounts
+  if (!accountsLoaded && !currentAccountMembershipId) {
+    return <ShowPageLoading message={t('Loading.Accounts')} />;
+  }
+
+  // If we have a selected account ID, redirect to that regardless of whether it
+  // actually exists. This help us show the correct error message on the Destiny
+  // page and avoid bouncing to another account when Bungie.net isn't returning
+  // all the accounts.
+  if (currentAccountMembershipId) {
+    return (
+      <Navigate to={accountRoute({ membershipId: currentAccountMembershipId, destinyVersion })} />
```

Comment:
This is a blind redirect - it doesn't matter if the account is known to us, it just sends you to the path for that account. This is to keep things stable even when Bungie.net is misbehaving.